service/handlers.go

```diff
@@ -0,0 +1,185 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/gorilla/mux"
+	"github.com/jackc/pgx/v4"
+	"github.com/mailgun/service/models"
+	"net/http"
+	"os"
+)
+
+const (
+	DomainThreshold  = 1000
+	BouncedThreshold = 1
+)
+
+type BaseHandler struct {
+	db *pgx.Conn
+}
+
+func NewBaseHandler(db *pgx.Conn) *BaseHandler {
+	return &BaseHandler{
+		db: db,
+	}
+}
+
+func (h *BaseHandler) DeliveredHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+
+		errResp, domainName := getDomain(r)
+
+		err := h.updateOrCreate(domainName, 1, 0)
+		if err != nil {
+			fmt.Println("Error occurred", err)
+			errResp = true
+		}
+
+		response := models.Response{
+			Message:      "Successfully updated domain name",
+			ResponseCode: http.StatusOK,
+			Error:        errResp,
+		}
+		js, err := json.Marshal(response)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		w.Header().Set("Content-Type", "application/json")
+		_, _ = w.Write(js)
```

Comment:
Should the error here be ignored?

Reply:
Correct. I can print out at least the error message here if we have write failures to debug the problem. Will push the fix